pkg/secrets/gcp_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package secrets
+
+import (
+	"encoding/base64"
+
+	"github.com/pkg/errors"
+	. "gopkg.in/check.v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	secerrors "github.com/kanisterio/kanister/pkg/secrets/errors"
+)
+
+type GCPSecretSuite struct{}
+
+var _ = Suite(&GCPSecretSuite{})
+
+func (s *GCPSecretSuite) TestValidateGCPCredentials(c *C) {
+	serviceAccountJson := make([]byte, base64.StdEncoding.EncodedLen(len([]byte("service_account_json"))))
+	base64.StdEncoding.Encode(serviceAccountJson, []byte("service_account_json"))
+	for _, tc := range []struct {
+		secret       *v1.Secret
+		errorChecker func(err error)
+	}{
+		{
+			secret: &v1.Secret{
+				Type: v1.SecretType(GCPSecretType),
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sec",
+					Namespace: "ns",
+				},
+				Data: map[string][]byte{
+					GCPProjectID:             []byte("key_id"),
+					GCPServiceAccountJsonKey: serviceAccountJson,
+				},
+			},
+			errorChecker: func(err error) {
+				c.Check(err, IsNil)
+			},
+		},
+		{ // Incomatible secret type
+			secret: &v1.Secret{
+				Type: v1.SecretType(AWSSecretType),
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "sec",
+					Namespace: "ns",
+				},
+				Data: map[string][]byte{
+					GCPProjectID:             []byte("key_id"),
+					GCPServiceAccountJsonKey: serviceAccountJson,
+				},
+			},
+			errorChecker: func(err error) {
+				c.Check(err, NotNil)
+				expectedError := errors.Wrapf(secerrors.ErrValidate, secerrors.IncompatibleSecretTypeErrorMsg, GCPSecretType, "ns", "sec")
+				c.Check(err, ErrorMatches, expectedError.Error())
```

Comment:
Instead of using `func()` as errorChecker and validating it against all the test cases. Should we use `expectedError` and `expectedErrorMsg` separate fields and validate those? It will reduce number of line (as we are repeating `c.Check()` for each test case as well as improve readability